services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -450,4 +450,65 @@ async def push_output_ports(
     return Response(status_code=status.HTTP_204_NO_CONTENT)
 
 
+@containers_router.post(
+    "/containers:restart",
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+    responses={
+        status.HTTP_404_NOT_FOUND: {"description": "Container does not exist"},
+        status.HTTP_422_UNPROCESSABLE_ENTITY: {
+            "description": "Error while running docker-compose command"
+        },
+    },
+)
+async def restarts_containers(
+    command_timeout: float = Query(
+        10.0, description="docker-compose stop command timeout default"
+    ),
+    settings: DynamicSidecarSettings = Depends(get_settings),
+    shared_store: SharedStore = Depends(get_shared_store),
+    rabbitmq: RabbitMQ = Depends(get_rabbitmq),
+) -> Response:
+    """Removes the previously started service
+    and returns the docker-compose output"""
+
+    stored_compose_content = shared_store.compose_spec
+    if stored_compose_content is None:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            detail="No spec for docker-compose command was found",
+        )
+
+    async def _run_command(command: str, command_timeout: Optional[float]) -> None:
+        finished_without_errors, stdout = await write_file_and_run_command(
+            settings=settings,
+            file_content=stored_compose_content,
+            command=command,
+            command_timeout=command_timeout,
+        )
+        if not finished_without_errors:
+            error_message = (f"'{command}' finished with errors\n{stdout}",)
+            logger.warning(error_message)
+            raise HTTPException(status.HTTP_422_UNPROCESSABLE_ENTITY, detail=stdout)
+
+    await _run_command(
+        command=(
+            "docker-compose --project-name {project} --file {file_path} "
+            "stop --timeout {stop_and_remove_timeout}"
+        ),
+        command_timeout=command_timeout,
+    )
+
+    await _run_command(
+        command="docker-compose --project-name {project} --file {file_path} start",
+        command_timeout=command_timeout,
+    )
+
+    await _send_message(rabbitmq, "Service was restarted please reload the UI")
+    await rabbitmq.send_event_reload_iframe()
```

Comment:
what is the difference between these 2 calls? they appear to be redundant

Reply:
The first will leave a trace in the "Logger". The second one will only display a short lived message.

Reply:
what do you mean a trace in the "logger"? this is going to rabbitmq log exchange right?
why are the signature completely different?

Reply:
The `_send_message` is a utility function which also logs in the dynamic-sidecar what is sent to the UI.
Wen I reference the logger I mean the "Logs" tab in the UI. A message will be present in the "Logs" tab and one will be displayed as a popup.

![Screenshot 2021-11-23 at 12 50 27](https://user-images.githubusercontent.com/5694077/143019256-a91bb8bc-1265-4ded-9997-17ae8a3792dd.png)

This is from the logs
![Screenshot 2021-11-23 at 12 51 17](https://user-images.githubusercontent.com/5694077/143019337-90d58348-e7f9-47f6-a99c-640fb443101e.png)